### PR TITLE
Add typing to methods of ``PyLinter`` and related functions

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

### CODE REVIEW

File: pylint/lint/pylinter.py [hunk 1]
```diff
@@ -460,10 +460,8 @@ def enable_fail_on_messages(self) -> None:
                         # message starts with a category value, flag (but do not enable) it
                         self.fail_on_symbols.append(msg.symbol)
 
-    def any_fail_on_issues(self):
-        return self.stats and any(
-            x in self.fail_on_symbols for x in self.stats.by_msg.keys()
-        )
+    def any_fail_on_issues(self) -> bool:
+        return any(x in self.fail_on_symbols for x in self.stats.by_msg.keys())
```
Comment (Pierre-Sassoulas) [inline pylint/lint/pylinter.py:464]: Is self.stats always initialized now (even for lib maintainer) ?

File: pylint/checkers/__init__.py [hunk 2]
```diff
@@ -128,7 +132,7 @@ def table_lines_from_stats(
     return lines
 
 
-def initialize(linter):
+def initialize(linter: PyLinter) -> None:
     """Initialize linter with checkers in this package."""
     register_plugins(linter, __path__[0])
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/__init__.py:137]: Not related to the current MR but it looks a little too magical, maybe we should use hard coded values ?

File: pylint/lint/pylinter.py (hunk 1 above)
Comment (DanielNoord) [reply pylint/lint/pylinter.py:464]: Yes 😄 

Finally some benefits from may refactors 😄

File: pylint/checkers/__init__.py (hunk 2 above)
Comment (DanielNoord) [reply pylint/checkers/__init__.py:137]: ?

It works kinda nicely as we don't need to add any new checkers when we add them.

Comment (Pierre-Sassoulas) [reply pylint/checkers/__init__.py:137]: Well, I'm saying that because I don't understand how it works 😄

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**